### DISCUSSION TITLE
How can I await a trpc query in the next.js pages dir

### DISCUSSION THREAD

**Discussion by FleetAdmiralJakob** [trpc/trpc#5255] [Q&A] [answered]
Hi, I want to await a trpc query in my next.js pages dir.

Something like this:
```
                    const city = await api.reverseGeoRouter.getCity.query({
                        lat: latitude,
                        lon: longitude,
                        });
```

**Comment by dellorogiulio** [comment]
Server or client side?
With the @next version of trpc,
Client side:
 you call call [useUtils] (https://trpc.io/docs/client/react/useUtils) and fetch the data

Server side [see here](https://trpc.io/docs/server/server-side-calls)


Is what are you looking for?

**Reply by FleetAdmiralJakob** [reply]
pages dir. so I want to do it client side.

**Comment by FleetAdmiralJakob** [comment, marked as answer]
Nevermind. I just used onSuccess and it worked client side
